site/js/ledAPI.js

```diff
@@ -0,0 +1,256 @@
+            deviceID = this.id;
+
+            /* update currentDeviceLabel shown to user */
+            $('#currentDeviceLabel').html(this.value);
+            document.getElementById('currentDeviceLabel').value =
+                this.value;
+            console.info("deviceID set to: ", deviceID);
+            $('#currentDeviceLabel').show();
+
+
+            /* If selected device is 'new', get device settings from server and update page */
+            if (lastDevice != deviceID) {
+                console.debug("lastDevice != deviceID ---> " +
+                    lastDevice + " != " + deviceID);
+
+                $apiUtils.getData(apiUrl, deviceID,
+                    updateMainContent, disableMainContent);
+                $('#brightness').toggleClass('look-disabled', false);
+
+                if (lastDevice && document.getElementById(
+                        lastDevice).classList.contains(
+                        'btn-current')) {
+                    document.getElementById(lastDevice).classList.remove(
+                        'btn-current');
+                }
+            }
+
+            console.debug("setting device button to current");
+            document.getElementById(deviceID).classList.add(
+                'btn-current');
+        });
+
+
+        /* Change name of device when form is submitted */
+        $('form').on('submit', function changeDeviceName(event) {
+
+            /* Prevent the page from reloading after form submit */
+            event.preventDefault();
+
+            /* Set the text-output span to the value of the first input */
+            var $input = $(this).find('input');
+            var input = $input.val();
+
+            /* update device name on server */
+            $apiUtils.putRequest(apiUrl, deviceID, 'name', input);
+
+            /*  update displayed name for device button */
+            $('#currentDeviceLabel').html(input.slice(0,
+                nameMaxLength));
+            document.getElementById(deviceID).value = input.slice(0,
+                nameMaxLength);
+
+        });
+
+    }
+
+    function updateOnButton(device) {
+        console.debug("updateOnButton: state received: ", device[1][
+            'onState'
+        ]);
+        switch (device[1]['onState']) {
+            case "true":
+                $("#onSwitch").prop("checked", true);
+                break;
+            case "false":
+                $("#onSwitch").prop("checked", false);
+                break;
+        }
+    }
+
+
+    function updateBrightSlider(device) {
+        var brightVal = device[1]['brightness']
+        $('#brightness-value').html(brightVal);
+        $('#slider-brightness').val(brightVal);
+    }
+
+    function updateDeviceList(newDevices) {
+        var remove = [];
+        var add = [];
+
+        newDevices.sort();
+
+        /* Identify old devices no longer online and remove */
+        for (var i in devices) {
```

Comment:
I'm assuming `devices` is an `Array` here.
```suggestion
        for (var i of devices) {
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
vs 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

Reply:
yes. :) thanks for pointing out.